src/prpy/tsr/generic.py

```diff
@@ -0,0 +1,275 @@
+import numpy
+from prpy.tsr.tsrlibrary import TSRFactory
+from prpy.tsr.tsr import TSR, TSRChain
+
+def get_manip_idx(robot, manip=None):
+    ''' 
+    Helper function for getting the manipulator index
+    to be used by the TSR
+    '''
+    if manip is None:
+        manip_idx = robot.GetActiveManipulatorIndex()
+    else:
+        import openravepy
+        p = openravepy.KinBody.SaveParameters
+        with robot.CreateRobotStateSaver(p.ActiveManipulator):
+            robot.SetActiveManipulator(manip)
+            manip_idx = manip.GetRobot().GetActiveManipulatorIndex()
+    return manip_idx
+
+def cylinder_grasp(robot, obj, obj_radius, obj_height, 
+                   lateral_offset = 0.0, 
+                   vertical_tolerance = 0.02,
+                   yaw_range = None,
+                   manip = None, **kwargs):
+    """
+    NOTE: This function makes the following assumptions:
+      1. The end-effector is oriented such that the z-axis is out of the palm
+          and the x-axis should be perpendicular to the object
+      2. The object coordinate frame is at the bottom, center of the object
+          
+    @param robot The robot performing the grasp
+    @param obj The object to grasp
+    @param obj_radius The radius of the object
+    @param obj_height The height of the object
+    @param lateral_offset The lateral offset from the edge of the object
+      to the end-effector
+    @param vertical_tolerance The maximum vertical distance from the vertical center
+      of the object that the grasp can be performed
+    @param yaw_range Allowable range of yaw around object (default: [-pi, pi])
+    @param manip The manipulator to perform the grasp, if None
+       the active manipulator on the robot is used
+    """
```

Comment:
Add bounds checking for `obj_radius`, `obj_height`, `vertical_tolerance`, and `yaw_range`.